pl.c

```diff
@@ -459,6 +452,41 @@ static void pl_delete_selected_pl(void)
 	job_schedule_pl_delete(pdd);
 }
 
+static void pl_delete_selected_pl(void)
+{
+	if (yes_no_query("Delete selected playlist? [y/N]") != UI_QUERY_ANSWER_YES)
+		return;
+
+	pl_delete(pl_visible);
+}
+
+void pl_delete_by_name(char *name)
+{
+	struct playlist *pl;
+	list_for_each_entry(pl, &pl_head, node) {
+		if (strcmp(pl->name, name) == 0) {
+			pl_delete(pl);
+			return;
+		}
+	}
+	error_msg("couldn't find a playlist named '%s' to delete", name);
+}
+
+void pl_delete_all(void)
+{
+	struct playlist *pl;
+	struct playlist *temp;
+	struct playlist *last = NULL;
+	list_for_each_entry_safe(pl, temp, &pl_head, node) {
+		if (list_len(&pl_head) == 1) {
+			last = pl;
+			break;
+		}
```

Comment:
Can this special case for the last item be removed now?

Reply:
I've tested the code without that block and it was working but I'm not entirely sure how the list functions with additions inside the loop so I didn't want to risk an infinite loop. I could remove it if it's fine. I removed the "last" variable with 68d9b0b for now.